app/controllers/items_controller.rb

```diff
@@ -24,6 +24,13 @@ class ItemsController < ApplicationController
     set_governing_apo
   ]
 
+  rescue_from Dor::Services::Client::UnexpectedResponse do |exception|
+    md = /\((.*)\)/.match exception.message
+    detail = JSON.parse(md[1])['errors'].first['detail']
+    redirect_to solr_document_path(params[:id]),
+                flash: { error: "Unable to retrieve the cocina model: #{detail.truncate(200)}" }
+  end
+
```

Comment:
would it be better for the DSC to have a specific error for validation failures?  Presumably with the DSA also having a specific error for validation failures?  Could be a separate issue.

Reply:
I'm not sure I understand DSC always returns UnexpectedResponse when there is a response that is not a 200 or 404 : https://github.com/sul-dlss/dor-services-client/blob/34645ee6759eeca7ff6191b8c40d478d4a7edc3c/lib/dor/services/client/object.rb#L52

DSA always returns an ErrorResponse model in these cases: https://github.com/sul-dlss/dor-services-app/blob/master/openapi.yml#L1023-L1034

Why only handle a subset of the errors when we can handle all of them?